### PR TITLE
Refactors atom/Initialize. Fixes ruins being initialized twice

### LINKED ISSUE #23133: Ruins are initialized twice
Once in SSobj/setup_template_objects and again in SSobj/Initialize

### PR DESCRIPTION
**Pull request by Cyberboss**
Captialized for compiling correctness and to be more inline with Destroy

Will now be called from atom/New if the world initialization loop in SSobj has already run. Should always call the base.

Fixes #23133

Now comes with the `roundstart` parameter (misnomer) indicating whether or not it was called by a map loading (main or runtime map) or by atom/New

Other fixes/tweaks:
- Renamed a proc called Initialize in abduction consoles to Setup
- Fixed the New call for /obj/item/device/radio/headset/headset_sec/alt/department
- Removed a spawn from the Initialize of turbine_computer which made literally zero sense
- Generalized the proc which fixes RND servers with no id set

Reasoning: It's better to check roundstart per function than to have to duplicate code in New and Initialize. Think of it as a safer New for atoms. If we move enough stuff to it, initial map load performance will increase due to less New calls

### CODE REVIEW

File: code/game/machinery/computer/atmos_control.dm [hunk 1]
```diff
@@ -44,14 +44,13 @@
...
+/obj/machinery/air_sensor/Initialize()
+	..()
 	set_frequency(frequency)
 
 /obj/machinery/air_sensor/New()
-	..()
 	SSair.atmos_machinery += src
-	if(SSradio)
-		set_frequency(frequency)
+	..()
```
Comment (optimumtact) [inline code/game/machinery/computer/atmos_control.dm]: why did you move this to the bottom? It's good practice in new to always have the parent call first, because that ensures you have a loc and other vars before your own code is running

Comment (Cyberboss) [inline code/game/machinery/computer/atmos_control.dm]: Slip of the whatever

**Comment by MrPerson** [comment]
I'm not against this in general but I just hope you understand this is going to have more of an impact than you might think at first.

**Comment by Cyberboss** [comment]
@optimumtact M8 you what? This is done

**Comment by optimumtact** [comment]
it's conflicting

**Comment by Cyberboss** [comment]
Merge conflicts? I see no merge conflicts.

File: code/game/atoms.dm [hunk 2]
```diff
@@ -431,8 +433,9 @@ var/list/blood_splatter_icons = list()
 //effects at world start up without causing runtimes
 /atom/proc/spawn_atom_to_world()
 
-//This will be called after the map and objects are loaded
-/atom/proc/initialize()
+//Called after New if the world is not loaded with TRUE
+//Called from base of New if the world is loaded with FALSE
+/atom/proc/Initialize(roundstart)
```
Comment (MrStonedOne) [inline code/game/atoms.dm]: World init speed up idea:

Make this a set waitfor = 0 proc so sleeps in here don't slow down initializing anything else.

**Comment by Cyberboss** [comment]
@optimumtact You are soooo fast with the WIP tag, get rid of it

**Comment by Cyberboss** [comment]
Testaruuuu Mergaruuuuuuuuuuuuuu!!!

**Comment by MrStonedOne** [comment]
If you're gonna beg for test merges and not be in irc to see when they happen don't be shocked when you find out your shit's been test merged 3 times now.

**Comment by Cyberboss** [comment]
I need to check out that issue, hold off on merging it for a sec

**Comment by Cyberboss** [comment]
Add `Fix` and should be good to merge

**Comment by MrStonedOne** [comment]
hold off on merging this until i figure out if iut causes https://github.com/tgstation/tgstation/issues/23070#issuecomment-274169413

**Comment by Cyberboss** [comment]
Any update?

**Comment by MrStonedOne** [comment]
This is good.

**Comment by AnturK** [comment]
Rename roundstart to something more reasonable and this is good to go.

**Comment by Cyberboss** [comment]
Reeeeeeeee

File: code/modules/holodeck/computer.dm [hunk 3]
```diff
@@ -58,15 +58,15 @@
 		world.log << "Holodeck computer cannot be in a holodeck."
 		world.log << "This would cause circular power dependency."
 		qdel(src)  // todo handle constructed computers
-		return
+		return	//l-lewd...
```
Comment (optimumtact) [inline code/modules/holodeck/computer.dm]: stop merging this shit

**Comment by Cyberboss** [comment]
I put that there because the code was fucky, actually, and I can't remember
why
